File: app/src/main/java/com/example/localservicemarketplace/models/Payment.java
package com.example.localservicemarketplace.models;

import java.math.BigDecimal;
import java.util.Date;
import java.util.UUID;

public class Payment {
    private String paymentId;
    private String bookingId;
    private BigDecimal amount;
    private PaymentStatus status;
    private PaymentMethod method;
    private String transactionRef;
    private Date paidAt;
    private Date createdAt;

    public enum PaymentStatus {
        PENDING,
        CONFIRMED,
        COMPLETED,
        CANCELED,
        REFUNDED
    }

    public enum PaymentMethod {
        CARD,
        WALLET,
        CASH_ON_SERVICE,
        OTHER
    }

    public Payment() {
        this.paymentId = UUID.randomUUID().toString();
        this.createdAt = new Date();
        this.status = PaymentStatus.PENDING;
    }

    // Getters and Setters
    public String getPaymentId() { return paymentId; }
    public void setPaymentId(String paymentId) { this.paymentId = paymentId; }
    public String getBookingId() { return bookingId; }
    public void setBookingId(String bookingId) { this.bookingId = bookingId; }
    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }
    public PaymentStatus getStatus() { return status; }
    public void setStatus(PaymentStatus status) { this.status = status; }
    public PaymentMethod getMethod() { return method; }
    public void setMethod(PaymentMethod method) { this.method = method; }
    public String getTransactionRef() { return transactionRef; }
    public void setTransactionRef(String transactionRef) { this.transactionRef = transactionRef; }
    public Date getPaidAt() { return paidAt; }
    public void setPaidAt(Date paidAt) { this.paidAt = paidAt; }
    public Date getCreatedAt() { return createdAt; }
    public void setCreatedAt(Date createdAt) { this.createdAt = createdAt; }
}

File: lib/domain/entities/payment.dart
import 'package:equatable/equatable.dart';
import 'package:localservicemarket/domain/enums/payment_method.dart';
import 'package:localservicemarket/domain/enums/payment_status.dart';

class Payment extends Equatable {
  const Payment({
    required this.paymentId,
    required this.bookingId,
    required this.amount,
    required this.status,
    required this.method,
    this.transactionRef,
    this.paidAt,
  });

  final String paymentId;
  final String bookingId;
  final double amount;
  final PaymentStatus status;
  final PaymentMethod method;
  final String? transactionRef;
  final DateTime? paidAt;

  Payment copyWith({
    String? paymentId,
    String? bookingId,
    double? amount,
    PaymentStatus? status,
    PaymentMethod? method,
    String? transactionRef,
    DateTime? paidAt,
  }) {
    return Payment(
      paymentId: paymentId ?? this.paymentId,
      bookingId: bookingId ?? this.bookingId,
      amount: amount ?? this.amount,
      status: status ?? this.status,
      method: method ?? this.method,
      transactionRef: transactionRef ?? this.transactionRef,
      paidAt: paidAt ?? this.paidAt,
    );
  }

  @override
  List<Object?> get props =>
      [paymentId, bookingId, amount, status, method, transactionRef, paidAt];
}
